mmdet/apis/ote/apis/detection/openvino_task.py

```diff
@@ -192,20 +129,21 @@ class OpenVINODetectionTask(IInferenceTask, IEvaluationTask, IOptimizationTask):
     def __init__(self, task_environment: TaskEnvironment):
         self.task_environment = task_environment
         self.model = self.task_environment.model
-        self.confidence_threshold: float = 0.0
+        self.hparams = self.task_environment.get_hyper_parameters(OTEDetectionConfig)
+        try:
+            self.confidence_threshold = np.frombuffer(self.model.get_data("confidence_threshold"), dtype=np.float32)[0]
+            self.hparams.inference_parameters.postprocessing.confidence_threshold = self.confidence_threshold
+        except KeyError:
+            self.confidence_threshold = self.hparams.inference_parameters.postprocessing.confidence_threshold
+        self.model_name = task_environment.model_template.name
         self.inferencer = self.load_inferencer()
 
-    @property
-    def hparams(self):
-        return self.task_environment.get_hyper_parameters(OTEDetectionConfig)
-
```

Comment:
Just can't understand, what you want to achieve here. Is it necessary to change to logic of the task to make in deployable? If not, please revert the changes related to confidence threshold.

Reply:
I define confidence threshold in __init__. But for Inferencer I use hparams, because having space for other parameters, not only conf threshold. I revert hparams as read-only property.

Reply:
hparams has a limited scope -- this is for user interaction mostly. Getting them from task environment guarantee to provide latest values. Please revert.